bigquery/google/cloud/bigquery/client.py

```diff
@@ -667,6 +668,79 @@ def query_rows(self, query, job_config=None, job_id=None, timeout=None):
         job.begin()
         return job.result(timeout=timeout)
 
+    def list_rows(self, table, selected_fields=None,
+                  max_results=None, page_token=None, start_index=None):
+        """List the rows of the table.
+
+        See
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/tabledata/list
+
+        .. note::
+
+           This method assumes that the provided schema is up-to-date with the
+           schema as defined on the back-end: if the two schemas are not
+           identical, the values returned may be incomplete. To ensure that the
+           local copy of the schema is up-to-date, call ``client.get_table``.
+
+        :type table: One of:
+                     :class:`~google.cloud.bigquery.table.Table`
+                     :class:`~google.cloud.bigquery.table.TableReference`
+        :param table: the table to list, or a reference to it.
+
+        :type selected_fields: list of :class:`SchemaField`
+        :param selected_fields:
+            The fields to return. Required if ``table`` is a
+            :class:`~google.cloud.bigquery.table.TableReference`.
+
+        :type max_results: int
+        :param max_results: (Optional) Maximum number of rows to return.
```

Comment:
Since this returns an iterator which automatically fetches the next page, this is more accurately described as the batch size.

Reply:
I rewrote the doc, but mentioned pages instead of saying "batch size".